module/otp.py

```diff
@@ -0,0 +1,95 @@
+"""
+    OTP module.
+"""
+import os
+import random
+import smtplib
+from email.mime.text import MIMEText
+
+from module.data import MAIL_TEMPLATE
+
+#: Define the following in settings.yaml
+HOST = 'HOST'
+USER = "USER"
+PASSWORD = "PASSWORD"
+
+class OTPException(Exception):
+    pass
+
+class OTPConnectionFailed(OTPException):
+    pass
+
+class OTPLoginFailed(OTPException):
+    pass
+
+class OTPNotAuthenticated(OTPException):
+    pass
+
+class OTPCodeNotSent(OTPException):
+    pass
+
+class OTPSender:
+    """
+        OTP sender class.
+
+        Attributes:
+            authenticated (bool): True if the user is authenticated, False otherwise.
+
+        Methods:
+            login() -> None: 
+                Logs in to the SMTP server. Modifies the authenticated attribute.
+            send(to: str) -> str | None:
+                Sends a randomly generated OTP to the specified email address.
+                Returns the OTP sent to the specified email address.
```

Comment:
This section is not needed nor indicated in the [google style guide](https://google.github.io/styleguide/pyguide.html), since it will be inferred by the docstrings under each class method